dozer-ingestion/aerospike/src/connector.rs

```diff
@@ -158,105 +350,218 @@ impl Connector for AerospikeConnector {
-        HttpServer::new(move || {
-            App::new()
-                .app_data(web::Data::new(State {
-                    tables_map: tables_map.clone(),
-                    tables_idx: tables_idx.clone(),
-                    ingestor: i.clone(),
-                }))
-                .service(healthcheck)
-                .service(event_request_handler)
-        })
-        .bind("127.0.0.1:5929")
-        .unwrap()
-        .run()
-        .await
-        .unwrap();
+            .await?;
+
+        let tables_index_map: HashMap<String, TableIndexMap> = mapped_schema
+            .into_iter()
+            .enumerate()
+            .map(|(table_index, schema)| {
+                let columns_map: HashMap<String, (usize, FieldType)> = schema
+                    .expect("Schema should be present")
+                    .schema
+                    .fields
+                    .iter()
+                    .enumerate()
+                    .map(|(i, field)| (field.name.clone(), (i, field.typ)))
+                    .collect();
+
+                (
+                    tables[table_index].name.clone(),
+                    TableIndexMap {
+                        table_index,
+                        columns_map,
+                    },
+                )
+            })
+            .collect();
+
+        let server_state = ServerState {
+            tables_index_map: tables_index_map.clone(),
+            ingestor: ingestor.clone(),
+        };
+
+        let _server = self.start_server(server_state)?.await;
 
         Ok(())
     }
 }
 
-async fn process_event(
+async fn map_event(
     event: AerospikeEvent,
-    tables_map: HashMap<String, HashMap<String, usize>>,
-    tables_idx: HashMap<String, usize>,
-    ingestor: Ingestor,
-) {
-    let table_name = event.key.get(1).unwrap().clone().unwrap();
-    if let Some(columns_map) = tables_map.get(table_name.as_str()) {
+    tables_map: HashMap<String, TableIndexMap>,
+) -> Result<Option<IngestionMessage>, AerospikeConnectorError> {
+    let key = event.key.clone();
+    if key.len() != 4 {
+        return Err(AerospikeConnectorError::InvalidKeyValue(key.clone()));
+    }
+
+    let set_name = match key.clone().get(1) {
+        None => Err(AerospikeConnectorError::NoSetNameFindInKey(key.clone())),
+        Some(Some(set_name)) => Ok(set_name.clone()),
+        Some(None) => Err(AerospikeConnectorError::SetNameIsNone(key.clone())),
+    }?;
+
+    if let Some(TableIndexMap {
+        columns_map,
+        table_index,
+    }) = tables_map.get(set_name.as_str())
+    {
         let mut fields = vec![Field::Null; columns_map.len()];
-        if let Some(pk) = columns_map.get("PK") {
-            fields[*pk] = match event.key.last().unwrap() {
-                None => Field::Null,
-                Some(s) => Field::String(s.to_string()),
-            };
+        if let Some((pk, _)) = columns_map.get("PK") {
+            fields[*pk] = key
+                .clone()
```

Comment:
Because you checked the key's length above, I think this whole block can just become
```rust
fields[*pk] = Field::String(key.remove(3));
```
which doesn't allocate at all.

I think the `PK` field will always be filled, right? If it's not, the message is not well-formed and we should error, I think. Anyway, downstream we can't handle messages without a PK